contents/tutorials/sticky-feature-flags.md

```diff
@@ -0,0 +1,208 @@
+---
+title: How to create sticky feature flags
+date: 2025-06-16
+author:
+  - dylan-martin
+showTitle: true
+sidebar: Docs
+tags:
+  - feature flags
+  - experiments
+---
+
+Sticky feature flags ensure users maintain their original variant assignment even after rollout conditions change. This is crucial for experiments where you need consistent user experiences throughout the test period, regardless of changes to targeting rules or rollout percentages.
+
+This tutorial shows how to set up sticky feature flags in PostHog by combining person properties with feature flag conditions to guarantee consistent variant assignments.
+
+## The problem with regular feature flags
+
+Regular feature flags can reassign users to different variants when you modify rollout conditions. For example:
+
+- You start an A/B experiment with 50% of users seeing variant A
+- Later, you decrease it to 25% rollout
+- Some users who previously saw variant A might now see variant B
+
+This inconsistency can skew your experiment results and create a poor user experience.
+
+## How sticky flags work
+
+Sticky flags solve this by:
+
+1. **Setting a person property** when a user first evaluates the flag
+2. **Adding override conditions** to the flag that check for existing assignments
+3. **Guaranteeing consistency** by prioritizing the stored assignment over new rollout rules
+
+## Setting up sticky feature flags
+
+### Step 1: Create your feature flag
+
+First, create your feature flag normally in PostHog:
+
+1. Go to [Feature Flags](https://app.posthog.com/feature_flags) and click "New feature flag"
+2. Set your flag key (e.g., `sticky-experiment`)
+3. Configure your initial rollout conditions
+4. Save the flag
+
+### Step 2: Capture person properties on evaluation
+
+When evaluating the flag in your code, capture the result as a person property. Here's an example using the JavaScript SDK:
+
+**Important:** Use `$set_once` instead of `$set` to ensure the property is only set on the first evaluation and never overwritten.
+
+```js
+// Check the feature flag
+const flagValue = posthog.isFeatureEnabled('sticky-experiment')
+
+// Capture the assignment as a person property
+posthog.capture('sticky_flag_evaluated', {
+  $set_once: { 
+    sticky_experiment_variant: flagValue ? 'test' : 'control' 
+  }
+})
+
+// Use the flag value in your code
+if (flagValue) {
+  // Show test variant
+} else {
+  // Show control variant  
+}
+```
```

Comment:
If they're trying to set up an experiment, I would expect the flag to return a variant. Something like:

```js
// Check the multivariate feature flag for the variant value
const flagValue = posthog.getFeatureFlag('sticky-experiment')

// Capture the assignment as a person property
posthog.capture('sticky_flag_evaluated', {
  $set_once: { 
    sticky_experiment_variant: flagValue 
  }
})

// Use the flag value in your code
if (flagValue === 'test') {
  // Show test variant
} else {
  // Show control variant  
}
```

Maybe it makes sense to show both code examples, one if you're doing a boolean flag, another if you're doing a multivariate flag?

Reply:
oh fuck I actually had that in at some point and then must've nuked it (I changed some examples).  I'll add!